php/src/TheClass.php

```diff
@@ -1,8 +1,8 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace Kata;
 
-class TheClass
+final class TheClass
```

Comment:
Is this required? Is meant to be the class you want to rewrite for the kata.

Reply:
The fact that this class is `final` or not it doesn't mean that the class shouldn't be changed. Defining your classes `final` by default has some interesting benefits. I already wrote an article about this topic, check it out: https://chemaclass.medium.com/final-classes-in-php-9174e3e2747e

Reply:
I agree about the benefits but IMHO this kind of decisions should be taken by the kata players instead of be pre-made in the template.

Reply:
I think it's good to introduce these little things & details even in some katas templates. Especially considering the scope of the template itself, which is basically a dummy class with some boilerplate code that has to be changed.
My point is that I see no downside of having this `final` here.

Actually, for some people who might don't about the existence of this `final` keyword, it might be a perfect opportunity to explain to them what's the meaning of it. What a better place to learn about this stuff than in a kata :)

But... well, if that is blocking the PR I can simply remove it (the `final` keyword) and merge the rest. We could discuss/think about the `final` in a follow-up PR if necessary. What do you think?

Reply:
If it is true that it can be interesting for people who don't know the  purpose of `final`, but it has very little impact on this kata, so it will surely go completely unnoticed.

To introduce concepts like `final` it is better to do it in katas that are completely oriented to do it.

Besides that it stains the rest of the PR that if I think they seem good changes.

Reply:
Allright. I just pushed a commit removing the final keywords. Win/win :)